Segfault When SHA resolution fails

<!--- Provide a general summary of the issue in the Title above -->

## Description & Reproduction
if env is set like SHA=""

this happens:

```
2023-10-06 05:24:01 ERR Couldn't extract git info for commit sha please set 'SHA' environment variable. process=main
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1033fee]

goroutine 1 [running]:
github.com/bearer/bearer/internal/report/output/saas.SendReport({0xc0010d7e00, {0x8bb2c97000, 0x12a05f200, 0x6fc23ac00, 0x3e8, 0xdf8475800, 0x1e8480, {0x0, 0x0}}, {{0x7fffa0e9ce64, ...}, ...}, ...}, ...)
	/github/workspace/internal/report/output/saas/saas.go:88 +0x32e
github.com/bearer/bearer/internal/report/output.GetData({{_, _}, _}, {0xc0010d7e00, {0x8bb2c97000, 0x12a05f200, 0x6fc23ac00, 0x3e8, 0xdf8475800, 0x1e8480, ...}, ...}, ...)
	/github/workspace/internal/report/output/output.go:70 +0x538
github.com/bearer/bearer/internal/commands/artifact.(*runner).Report(0xc0014a98c0, {0x0?, 0x0?, 0x0?}, 0x0?)
	/github/workspace/internal/commands/artifact/run.go:379 +0x269
github.com/bearer/bearer/internal/commands/artifact.Run({_, _}, {{{0x0, 0x0}, {0x0, 0x0}, {0x0, 0x0}}, {{0x0, 0x0}, ...}, ...})
	/github/workspace/internal/commands/artifact/run.go:325 +0x7e6
github.com/bearer/bearer/internal/commands.NewScanCommand.func2(0xc000005200, {0xc00008ddb0?, 0x1, 0x5})
	/github/workspace/internal/commands/scan.go:89 +0x398
github.com/spf13/cobra.(*Command).execute(0xc000005200, {0xc00008dd60, 0x5, 0x5})
	/github/home/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:940 +0x87c
github.com/spf13/cobra.(*Command).ExecuteC(0xc000004900)
	/github/home/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:1068 +0x3a5
github.com/spf13/cobra.(*Command).Execute(0x196e5c8?)
	/github/home/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:992 +0x13
main.run()
	/github/workspace/cmd/bearer/main.go:18 +0x34
main.main()
	/github/workspace/cmd/bearer/main.go:11 +0x13
```

Other env vars like CURRENT_BRANCH might also have this issue

## Expected Behavior

exits cleanly

## Actual Behavior
segfault

## Possible Fix

check return from sha check

## Your Environment

- Output of 'bearer version':
```
bearer version: 1.25.0
```

--- a/internal/report/output/saas/saas.go
+++ b/internal/report/output/saas/saas.go
@@ -75,6 +75,7 @@ func SendReport(config settings.Config, reportData *types.ReportData) {
 			errorMessage := fmt.Sprintf("Unable to calculate Metadata. %s", err)
 			log.Debug().Msgf(errorMessage)
 			config.Client.Error = &errorMessage
+			return
 		}
 	}
 

